gwcs/geometry.py

```diff
@@ -107,35 +164,85 @@ class CartesianToSpherical(Model):
     Spherical coordinates are assumed to be in degrees with ``phi`` being
-    the *azimuthal* angle ``[0, 360)`` and ``theta`` being the *elevation*
-    angle ``[-90, 90]``.
+    the *azimuthal angle* ``[0, 360)`` (or ``[-180, 180)``) and ``theta``
+    being the *elevation angle* ``[-90, 90]`` or the *inclination (polar)
+    angle* in range ``[0, 180]`` depending on the used definition.
 
     """
     _separable = False
 
     n_inputs = 3
     n_outputs = 2
 
-    def __init__(self, **kwargs):
+    def __init__(self, wrap_phi_at=360, theta_def='elevation', **kwargs):
+        """
+        Parameters
+        ----------
+        wrap_phi_at : {180, 360}, optional
+            Specifies the range of the azimuthal angle. When ``wrap_phi_at`` is
+            180, azimuthal angle will have a range of ``[-180, 180)`` and
+            when ``wrap_phi_at`` is 360 (default), the azimuthal angle will have
+            a range of ``[0, 360)``
+
+        theta_def : {'elevation', 'inclination', 'polar'}, optional
+            Specifies the definition used for the ``theta``: 'elevation' from
+            the reference plane or 'inclination' (or 'polar') angle measured
+            from the pole.
+
+        """
         super().__init__(**kwargs)
+        self.wrap_phi_at = wrap_phi_at
+        self.theta_def = theta_def
         self.inputs = ('x', 'y', 'z')
         self.outputs = ('phi', 'theta')
 
-    @staticmethod
-    def evaluate(x, y, z):
+    @property
+    def wrap_phi_at(self):
+        return self._wrap_phi_at
+
+    @wrap_phi_at.setter
+    def wrap_phi_at(self, wrap_angle):
+        if not _is_int(wrap_angle):
+            raise TypeError("'wrap_phi_at' must be an integer number: 180 or 360")
+        if wrap_angle not in [180, 360]:
+            raise ValueError("Allowed 'wrap_phi_at' values are 180 and 360")
+        self._wrap_phi_at = wrap_angle
+
+    @property
+    def theta_def(self):
+        return self._theta_def
+
+    @theta_def.setter
+    def theta_def(self, theta_def):
+        if not isinstance(theta_def, str):
+            raise TypeError("'theta_def' must be a string.")
+        if theta_def not in ['elevation', 'inclination', 'polar']:
+            raise ValueError("Allowed 'theta_def' values are: 'elevation', "
+                             "'inclination', or 'polar'.")
+        self._theta_def = theta_def
+
+    def evaluate(self, x, y, z):
         nquant = [isinstance(i, u.Quantity) for i in (x, y, z)].count(True)
         if nquant in [1, 2]:
             raise TypeError("All arguments must be of the same type "
                             "(i.e., quantity or not).")
 
         h = np.hypot(x, y)
-        phi = np.mod(
-            np.rad2deg(np.arctan2(y, x)),
-            360.0 * u.deg if nquant else 360.0
-        )
+        phi = np.rad2deg(np.arctan2(y, x))
+        if h == 0.0:
+            phi *= 0.0
+
+        if self._wrap_phi_at != 180:
+            phi = np.mod(phi, 360.0 * u.deg if nquant else 360.0)
```

Comment:
It's not obvious to me that if the input cartesian coordinates are quantities, the angles must be. I guess that is somewhat implied, but is this generally assumed in astropy?